pkg/scheduler/algorithmprovider/registry.go

```diff
@@ -0,0 +1,158 @@
+type Registry map[string]*Config
+
+// NewRegistry returns an algorithm provider registry instance.
+func NewRegistry(hardPodAffinityWeight int64) Registry {
+	registry := Registry{}
+	defaultConfig := getDefaultConfig(hardPodAffinityWeight)
+	applyFeatureGates(defaultConfig)
+	registry[schedulerapi.SchedulerDefaultProviderName] = defaultConfig
+
+	caConfig := getClusterAutoscalerConfig(hardPodAffinityWeight)
+	applyFeatureGates(caConfig)
+	registry[ClusterAutoscalerProvider] = caConfig
+	return registry
+}
+
+func getDefaultConfig(hardPodAffinityWeight int64) *Config {
+	return &Config{
+		FrameworkPlugins: &schedulerapi.Plugins{
+			PreFilter: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: noderesources.FitName},
+					{Name: nodeports.Name},
+					{Name: interpodaffinity.Name},
+				},
+			},
+			Filter: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: nodeunschedulable.Name},
+					{Name: noderesources.FitName},
+					{Name: nodename.Name},
+					{Name: nodeports.Name},
+					{Name: nodeaffinity.Name},
+					{Name: volumerestrictions.Name},
+					{Name: tainttoleration.Name},
+					{Name: nodevolumelimits.EBSName},
+					{Name: nodevolumelimits.GCEPDName},
+					{Name: nodevolumelimits.CSIName},
+					{Name: nodevolumelimits.AzureDiskName},
+					{Name: volumebinding.Name},
+					{Name: volumezone.Name},
+					{Name: interpodaffinity.Name},
+				},
+			},
+			PostFilter: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: interpodaffinity.Name},
+				},
+			},
+			Score: &schedulerapi.PluginSet{
+				Enabled: []schedulerapi.Plugin{
+					{Name: noderesources.BalancedAllocationName, Weight: 1},
+					{Name: imagelocality.Name, Weight: 1},
+					{Name: interpodaffinity.Name, Weight: 1},
+					{Name: noderesources.LeastAllocatedName, Weight: 1},
+					{Name: nodeaffinity.Name, Weight: 1},
+					{Name: nodepreferavoidpods.Name, Weight: 10000},
+					{Name: defaultpodtopologyspread.Name, Weight: 1},
+					{Name: tainttoleration.Name, Weight: 1},
+				},
+			},
+		},
+		FrameworkPluginConfig: []schedulerapi.PluginConfig{
+			{
+				Name: interpodaffinity.Name,
+				Args: runtime.Unknown{Raw: []byte(fmt.Sprintf(`{"hardPodAffinityWeight":%d}`, hardPodAffinityWeight))},
+			},
+		},
+	}
+}
+
+func getClusterAutoscalerConfig(hardPodAffinityWeight int64) *Config {
+	defaultConfig := getDefaultConfig(hardPodAffinityWeight)
+	caConfig := Config{
+		FrameworkPlugins: &schedulerapi.Plugins{},
+	}
+	defaultConfig.FrameworkPlugins.DeepCopyInto(caConfig.FrameworkPlugins)
+	caConfig.FrameworkPluginConfig = append([]schedulerapi.PluginConfig(nil), defaultConfig.FrameworkPluginConfig...)
+
+	// Replace least with most requested.
+	for i := range caConfig.FrameworkPlugins.Score.Enabled {
+		if caConfig.FrameworkPlugins.Score.Enabled[i].Name == noderesources.LeastAllocatedName {
+			caConfig.FrameworkPlugins.Score.Enabled[i].Name = noderesources.MostAllocatedName
+		}
+	}
```

Comment:
why are replacing the least with most requested?

Reply:
this is the existing characteristic of the Cluster Autoscaler provider: https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithmprovider/defaults/defaults.go#L101